crates/config/src/soldeer.rs

```diff
@@ -0,0 +1,22 @@
+//! Configuration specific to the `forge soldeer` command and the `forge_soldeer` package
+
+use serde::{ Deserialize, Serialize };
+
+/// Soldeer dependencies config structure
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct SoldeerDependency {
+    /// The version of the dependency
+    pub version: String,
+
+    /// The url from where the dependency was retrieved
+    pub url: String,
+}
```

Comment:
I think we want support for downloading from git, and generally shape this more like Cargo, but for now this is ok if we can make the URL optional (defaulting to the soldeer repo, if not possible let's discuss)

Some examples:

These would be the defaults

```toml
dep = "v1.0.0" # download from default repository (soldeers)
```

```toml
dep = { version = "v1.0.0" } # download from default repository (soldeers)
```

This is most likely needed for development when you depend on patches not yet released:

```toml
dep = { git = "https://github.com/foo/bar", ref = "deadbeef" } # fetch specific commit or tag from git repo (using libgit or similar)
```

```toml
dep = { git = "https://github.com/foo/bar", branch = "cool-feature" } # fetch specific branch from git repo (using libgit or similar)
```

Leaving out ref/branch would default to the default branch (master or main)

At some point when we lock in project structure we should also support `path` for monorepos, e.g.

```toml
dep = { path = "some/path/to/dep" } # replaces specifying remappings in monorepos
```

Finally, at some point we should also support using custom registries.

**To be clear** I want the first two to be in before we can merge this PR; the others should come later.

Reply:
git integration is not yet there, second one is possible, i need to see the first one if it works.

Reply:
It works also with this:
```
dep = "1.0.0" # download from default repository (soldeers)
```

I also pushed a test.